audio-echo/app/src/main/java/com/google/sample/echo/MainActivity.java

```diff
@@ -38,6 +46,8 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        //get run-time permission...
+        requestResourcePermission();
```

Comment:
this is good one: the authorization is a callback. I am not sure whether the callback is in the same thread or in different thread:  if it is in different thread, then we have to include a waiting if put this request inside the button click action handler.  felt this is easier and by requesting the permission during the app launch: for this app, if recording is not granted, we really could not do anything at all.